src/widgets/FormControl.js

```diff
@@ -81,8 +82,10 @@ const FormControlComponent = ({
     setRefs({ length: inputConfig.length });
   }, [isFocused]);
 
+  const debouncedUpdateForm = useDebounce(onUpdateForm, 500);
```

Comment:
Debouncing the updates for less dispatches- seems to work well except for the slight jank for the final validation